src/Collection/Resources/assets/src/controller.ts

```diff
@@ -0,0 +1,115 @@
+import { Controller } from '@hotwired/stimulus';
+
+const DEFAULT_ITEMS_SELECTOR = ':scope > :is(div, fieldset)';
+
+interface CollectionDataset extends DOMStringMap {
+    prototype: string;
+    currentIndex: string;
+    itemsSelector?: string;
+    addButton?: string;
+    deleteButton?: string;
+}
+
+enum ButtonType {
+    Add,
+    Delete,
+}
+
+export default class extends Controller {
+    static values = {
+        addButton: '',
+        deleteButton: '',
+    };
+
+    connect() {
+        this.connectCollection(this.element as HTMLElement);
+    }
+
+    connectCollection(parent: HTMLElement) {
+        parent.querySelectorAll('[data-prototype]').forEach((el) => {
+            const collectionEl = el as HTMLElement;
+            const items = this.getItems(collectionEl);
+            collectionEl.dataset.currentIndex = items.length.toString();
+
+            this.addAddButton(collectionEl);
+
+            this.getItems(collectionEl).forEach((itemEl) => this.addDeleteButton(collectionEl, itemEl as HTMLElement));
+        });
+    }
+
+    getItems(collectionElement: HTMLElement) {
+        return collectionElement.querySelectorAll(collectionElement.dataset.itemsSelector || DEFAULT_ITEMS_SELECTOR);
+    }
+
+    createButton(collectionEl: HTMLElement, buttonType: ButtonType): HTMLElement | null {
+        const attributeName = `${ButtonType[buttonType].toLowerCase()}Button`;
+        const button = collectionEl.dataset[attributeName] ?? (this.element as HTMLElement).dataset[attributeName];
+        console.log(button);
+
+        // Button explicitly disabled through data attribute
+        if ('' === button) return null;
+
+        // No data attribute provided or <template> not supported: create raw HTML button
+        if (undefined === button || !('content' in document.createElement('template'))) {
+            const button = document.createElement('button') as HTMLButtonElement;
+            button.type = 'button';
+            button.textContent = ButtonType[buttonType];
+
+            return button;
+        }
+
+        // Use the template referenced by the data attribute
+        const buttonTemplate = document.getElementById(button) as HTMLTemplateElement | null;
+        if (!buttonTemplate) throw new Error(`template with ID "${buttonTemplate}" not found`);
+
+        const fragment = buttonTemplate.content.cloneNode(true) as DocumentFragment;
+        if (1 !== fragment.children.length)
+            throw new Error('template with ID "${buttonTemplateID}" must have exactly one child');
```

Comment:
this needs to use backticks, not single quotes. Otherwise the interpolation won't work.